### PR TITLE
Feat/Expenses/RF32: Endpoint for expenseReport data by id

### PR DESCRIPTION
**Pull request by Ian326**
## Expenses-RF32: Endpoint for expenseReport data by id
ESTE PR ES EL QUE INCORRECTAMENTE SE HIZO A DEVELOP, SOLO QUE AHORA SE HARÁ A MBI

[Feature/RF32]: Added an endpoint for getting expenseReport data by id. Added base logic for expenses and all of it's corresponding files

### Description

RF32: As a user I want to consult the details of a travel expenses report.

### Changes

- Added `prisma/migrations/20240522045959_expenses_update_fields/migration.sql`. Migration to support corresponding fields for expenses according to updated ER Model
- Modified `prisma/schema.prisma` as a result of the previous migration
- Modified `src/api/routes/index.routes.ts`. Added endpoint for /expense/
- Added `src/api/routes/expense.routes.ts`
- Added `src/api/controllers/expense.controller.ts`
- Added `src/core/domain/entities/expense.entity.ts`
- Added `src/core/app/services/expense.service.ts`
- Added `src/core/infra/mappers/expense-entity-from-db-model.mapper.ts`
- Added `src/core/infra/repositories/expense.repository.ts`
- Modified `src/utils/enums/index.ts`. Added enum for ExpenseReportStatus
- Added `src/core/app/services/__tests__/expense.service.test.ts` for testing services

### Story Points

8

### Testing

A test was designed and run in which the service was tested. Mocha&Chai was used for this automated test. The test succeeds
image

### Dependencies

None (It will be the PR that lays the foundation for the entire expenses module)

### Done?

- [x] La tarea ha sido completamente implementada, cumpliendo con todos los requisitos y funcionalidades definidos.
- [x] Todas las pruebas pasan, siguiendo el plan de pruebas.
- [x] El código sigue los estándares de codificación y las mejores prácticas del proyecto.
- [x] La documentación ha sido actualizada (si corresponde).
- [x] Se han cerrado todos los issues (si aplicara) creados en el defect Log.
- [x] Se ha creado o actualizado la documentación necesaria relacionada con la tarea.
- [x] Se ha mapeado en las herramientas de trazabilidad [RTM](https://docs.google.com/spreadsheets/d/1Dqh-fTPP5tEfbQ_cVwz-jI74CmAe0DkSAw2a-vVrbVk/edit?usp=sharing), [PVG](https://docs.google.com/spreadsheets/d/1mFO9Bb7gJysk4sDaJ2z7ufl0epSR40XKp3J-r4IxFVc/edit#gid=1369995251)

### Screenshot
image

### CODE REVIEW

**Review by ArmandoRosasB — APPROVED** [review]

File: src/core/domain/entities/expense.entity.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+   */
+  supplier: string;
+  /**
+   * @param totalAmount: Decimal - Expense amount
+   */
+  totalAmount: Decimal;
+  /**
+   * @param status: string - Expense status
+   */
+  status?: string | null;
```
Comment (ArmandoRosasB) [inline src/core/domain/entities/expense.entity.ts:49]: Usar el enum

File: src/core/domain/entities/expense.entity.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+   */
+  start_date: Date;
+  /**
+   * @param endDate: Date - Expense report end date
+   */
+  end_date?: Date | null;
+  /**
+   * @param status: string - Expense report status
+   */
+  status?: string | null;
```
Comment (ArmandoRosasB) [inline src/core/domain/entities/expense.entity.ts:205]: Usar el enum

File: src/core/domain/entities/expense.entity.ts (hunk 1 above)
Comment (dembA7) [reply src/core/domain/entities/expense.entity.ts:49]: x2

File: src/core/domain/entities/expense.entity.ts (hunk 2 above)
Comment (dembA7) [reply src/core/domain/entities/expense.entity.ts:205]: x2

**Review by ArmandoRosasB — APPROVED** [review]

**Review by RamonaNF — APPROVED** [review]

**Review by dembA7 — APPROVED** [review]

**Pull request merged**